File: skellig-junit-runner/src/main/java/org/skellig/runner/tagextractor/RequestedTagExtractor.java
package org.skellig.runner.tagextractor;

import org.skellig.feature.TestPreRequisites;

import java.util.Collection;
import java.util.Optional;

public class RequestedTagExtractor implements TagExtractor {

    @Override
    public <T> Optional<T> extract(Class<T> tagClass, Collection<TestPreRequisites<?>> testPreRequisites) {
        return testPreRequisites.stream()
                .filter(item -> tagClass.equals(item.getDetails().getClass()))
                .map(item -> (T)item.getDetails())
                .findFirst();
    }
}

File: skellig-junit-runner/src/main/kotlin/org/skellig/runner/tagextractor/RequestedTagExtractor.kt
package org.skellig.runner.tagextractor

import org.skellig.feature.metadata.TestMetadata

class RequestedTagExtractor : TagExtractor {

    override fun <T> extract(tagClass: Class<T>, testMetadata: Collection<TestMetadata<*>>): T? {
        return testMetadata
                .filter { tagClass == it.getDetails()!!.javaClass }
                .map { it.getDetails() as T }
                .firstOrNull()
    }
}